### PR TITLE
Refactor site to remove jekyll-bootstrap, use midnight theme

### PR DESCRIPTION
**Pull request by chakru-r**
Tested via `jekyll serve` locally.
Layout templates will require more tweaks.
For all new posts  - need to decide on/standardise whether to use title (auto generated from post file name) vs tagline (explicitly being mentioned in the post) to avoid duplicate titles in the final page.

### CODE REVIEW

File: _layouts/default.html [hunk 1]
```diff
@@ -1,6 +1,28 @@
...
-{% include themes/bootstrap/default.html %}
+<!doctype html>
+<html lang="{{ site.lang | default: "en-US" }}">
+  <head>
+    <meta charset="utf-8">
+    <meta http-equiv="X-UA-Compatible" content="IE=edge">
+
+{% seo %}
+    <link rel="stylesheet" href="{{ '/assets/css/style.css?v=' | append: site.github.build_revision | relative_url }}">
+    <script src="https://code.jquery.com/jquery-1.12.4.min.js" integrity="sha256-ZosEbRLbNQzLpnKIkEdrPv7lOy9C27hHQ+Xp8a4MxAQ=" crossorigin="anonymous"></script>
```
Comment (hrj) [inline _layouts/default.html:9]: Why is jquery required?

File: _layouts/default.html [hunk 2]
```diff
@@ -1,6 +1,28 @@
...
+    <script src="{{ '/assets/js/respond.js' | relative_url }}"></script>
```
Comment (hrj) [inline _layouts/default.html:10]: What is this respond.js for?

File: index.md [hunk 3]
```diff
@@ -41,7 +41,7 @@ BAS also conducted a session on astrophotography at Amrita Institute of Technolo
 
 BAS conducted an outreach program for school children at Sri Venketeshwara Rural High School on
 24th February,2024 evening at Challakere,Karnataka. Pictures and a general description will be updated soon.
-[Link to Challakere event page](/_posts/2024-02-26-Challakere-event.markdown)
+[Link to Challakere event page]({% link _posts/2024-02-26-Challakere-event.markdown %})
```
Comment (hrj) [inline index.md:44]: What is this `link` for? Can't Jekyll handle simple markdown links automatically?

File: _config.yml [hunk 4]
```diff
@@ -24,79 +24,7 @@ author :
...
-  # Some JB helpers refer to these paths; change them here if needed.
-  #
-  archive_path: /archive.html
-  categories_path : /categories.html
-  tags_path : /tags.html
-  atom_path : /atom.xml
-  rss_path : /rss.xml
+remote_theme: pages-themes/midnight@v0.2.0
+plugins:
+- jekyll-remote-theme # add this line to the plugins list if you already have one
```
Comment (hrj) [inline _config.yml:29]: Can we avoid the remote theme by copying it locally? This will avoid issues similar to what we had with the bootstrap plugin.

**Review by hrj — CHANGES_REQUESTED** [review]
A few requests and clarifications.

File: _config.yml (hunk 4 above)
Comment (chakru-r) [reply _config.yml:29]: OK - will do, let me try that. One thing is the reference here is to a specific version - so we won't pull in new updates.

File: index.md (hunk 3 above)
Comment (chakru-r) [reply index.md:44]: I noticed this link was broken when I was testing locally - and saw [this](https://jekyllrb.com/docs/liquid/tags/#link) in jekyll documentation and was following it. Though, I see now the link is not broken in pages.bas.org.in. I dont know if this behaviour was affected due to bootstrap plugin being used (don't understand enough of it).

File: _layouts/default.html (hunk 2 above)
Comment (chakru-r) [reply _layouts/default.html:10]: Came with the theme - it is an asset that is part of the [theme](https://github.com/pages-themes/midnight/tree/master/assets/js)

File: _layouts/default.html (hunk 1 above)
Comment (chakru-r) [reply _layouts/default.html:9]: I should have explained this a bit - default.html is actually what comes with the theme with some modifications (based on [this](https://github.com/pages-themes/midnight/blob/master/README.md#layouts) recommendation. The theme included default template has a link to the github repo right on top and downloading it  etc. See this [sample](https://pages-themes.github.io/midnight/)
The changes I made is to override that default template to remove that top section.

File: _config.yml (hunk 4 above)
Comment (hrj) [reply _config.yml:29]: Nevermind. I just saw the documentation for pulling the remote theme into local, and it doesn't seem worth the hassle. Let's go with remote for now.

**Review by hrj — APPROVED** [review]

**Pull request merged**